### PR TITLE
dev-python/*: more py3.10 porting

### PR DESCRIPTION
**Pull request by jsmolic**

### CODE REVIEW

File: dev-python/html5lib/files/html5lib-1.1-pytest6.patch [hunk 1]
```diff
@@ -0,0 +1,151 @@
+From e6bd99e8f2497194ffd0a06c6954ebb28d7526bb Mon Sep 17 00:00:00 2001
```
Comment (thesamesam) [inline dev-python/html5lib/files/html5lib-1.1-pytest6.patch:1]: Add link to source and relevant bugs

File: dev-python/html5lib/html5lib-1.1.ebuild [hunk 2]
```diff
@@ -18,6 +18,10 @@ KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~mips ppc ppc64 ~riscv ~s390 sparc x
 IUSE="test"
 RESTRICT+=" !test? ( test )"
```
Comment (thesamesam) [inline dev-python/html5lib/html5lib-1.1.ebuild:19]: Remove both lines, det does it for us

File: dev-python/html5lib/html5lib-1.1.ebuild [hunk 3]
```diff
@@ -18,6 +18,10 @@ KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~mips ppc ppc64 ~riscv ~s390 sparc x
...
 
+PATCHES=(
+	"${FILESDIR}"/${P}-pytest6.patch
+)
+
 RDEPEND=">=dev-python/six-1.9[${PYTHON_USEDEP}]
 	dev-python/webencodings[${PYTHON_USEDEP}]"
 BDEPEND="
```
Comment (thesamesam) [inline dev-python/html5lib/html5lib-1.1.ebuild:27]: Drop adding RDEPEND in test as det does it for us

File: dev-python/wcwidth/wcwidth-0.2.5-r1.ebuild [hunk 4]
```diff
@@ -24,13 +24,21 @@ RDEPEND="
...
-
 src_prepare() {
 	sed -e 's:--cov-append::' \
 		-e 's:--cov-report=html::' \
 		-e 's:--cov=wcwidth::' \
 		-i tox.ini || die
 	sed -i -e 's:test_package_version:_&:' tests/test_core.py || die
+
+	# causes pytest to fail
+	sed -i '/^looponfailroots =/d' tox.ini || die
```
Comment (thesamesam) [inline dev-python/wcwidth/wcwidth-0.2.5-r1.ebuild:35]: Bug ref

**Comment by thesamesam** [comment]
Huge thanks for working on this and making a big start here.

File: dev-python/wcwidth/wcwidth-0.2.5-r1.ebuild (hunk 4 above)
Comment (jsmolic) [reply dev-python/wcwidth/wcwidth-0.2.5-r1.ebuild:35]: Yeah, makes sence, I've added the reference

File: dev-python/html5lib/files/html5lib-1.1-pytest6.patch (hunk 1 above)
Comment (jsmolic) [reply dev-python/html5lib/files/html5lib-1.1-pytest6.patch:3]: Done

File: dev-python/html5lib/html5lib-1.1.ebuild (hunk 2 above)
Comment (jsmolic) [reply dev-python/html5lib/html5lib-1.1.ebuild]: done

File: dev-python/html5lib/html5lib-1.1.ebuild (hunk 3 above)
Comment (jsmolic) [reply dev-python/html5lib/html5lib-1.1.ebuild:25]: done

File: dev-python/pgspecial/metadata.xml [hunk 5]
```diff
@@ -1,7 +1,14 @@
...
 <pkgmetadata>
-	<!-- maintainer-needed -->
+	<maintainer type="person" proxied="yes">
+		<email>jakov.smolic@sartura.hr</email>
+		<name>Jakov Smolic</name>
+	</maintainer>
+	<maintainer type="project" proxied="proxy">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
```
Comment (mgorny) [inline dev-python/pgspecial/metadata.xml:11]: Please add Python too.

File: dev-python/psycopg/psycopg-2.8.6-r1.ebuild [hunk 6]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{7..9} )
+PYTHON_COMPAT=( python3_{7..10} )
 
 inherit distutils-r1 flag-o-matic
```
Comment (mgorny) [inline dev-python/psycopg/psycopg-2.8.6-r1.ebuild:9]: PostgreSQL server dep should go into BDEPEND.

File: dev-python/click-default-group/click-default-group-1.2.2.ebuild [hunk 7]
```diff
@@ -3,7 +3,8 @@
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{7,8,9} pypy3 )
+DISTUTILS_USE_SETUPTOOLS=bdepend
```
Comment (mgorny) [inline dev-python/click-default-group/click-default-group-1.2.2.ebuild:6]: We don't spell out `bdepend` explicitly.

File: dev-python/random2/random2-1.0.1-r1.ebuild [hunk 8]
```diff
@@ -1,9 +1,9 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{7,8,9} )
+PYTHON_COMPAT=( python3_{7..10} )
 inherit distutils-r1
```
Comment (mgorny) [inline dev-python/random2/random2-1.0.1-r1.ebuild:7]: Redundant bdep on setuptools.

File: dev-python/bottle/bottle-0.12.19.ebuild [hunk 9]
```diff
@@ -1,9 +1,9 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{7,8,9} pypy3 )
+PYTHON_COMPAT=( python3_{7..10} pypy3 )
```
Comment (mgorny) [inline dev-python/bottle/bottle-0.12.19.ebuild:6]: Remove the empty RDEPEND.

File: dev-python/cloudpickle/cloudpickle-1.6.0.ebuild [hunk 10]
```diff
@@ -2,7 +2,8 @@
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
-PYTHON_COMPAT=( python3_{7..9} )
+
+PYTHON_COMPAT=( python3_{7..10} )
```
Comment (mgorny) [inline dev-python/cloudpickle/cloudpickle-1.6.0.ebuild:6]: `epytest -s` in this one.

File: dev-python/webencodings/webencodings-0.5.1-r1.ebuild [hunk 11]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{7,8,9} pypy3 )
+PYTHON_COMPAT=( python3_{7..10} pypy3 )
 
 inherit distutils-r1
```
Comment (mgorny) [inline dev-python/webencodings/webencodings-0.5.1-r1.ebuild:8]: `||die` after `cat`.

File: dev-python/u-msgpack/u-msgpack-2.7.1.ebuild [hunk 12]
```diff
@@ -2,7 +2,7 @@
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
-PYTHON_COMPAT=( python3_{7,8,9} pypy3 )
+PYTHON_COMPAT=( python3_{7..10} pypy3 )
 
 inherit distutils-r1
```
Comment (mgorny) [inline dev-python/u-msgpack/u-msgpack-2.7.1.ebuild:8]: S immediatly after SRC_URI.

File: dev-python/pytest-expect/pytest-expect-1.1.0-r1.ebuild [hunk 13]
```diff
@@ -2,7 +2,7 @@
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
-PYTHON_COMPAT=( python3_{7,8,9} pypy3 )
+PYTHON_COMPAT=( python3_{7..10} pypy3 )
```
Comment (mgorny) [inline dev-python/pytest-expect/pytest-expect-1.1.0-r1.ebuild:5]: Remove empty IUSE, and put RESTRICT there.

File: dev-python/versioneer/versioneer-0.19.ebuild [hunk 14]
```diff
@@ -1,10 +1,10 @@
...
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
 
 DISTUTILS_USE_SETUPTOOLS=rdepend
-PYTHON_COMPAT=( python3_{7..9} )
+PYTHON_COMPAT=( python3_{7..10} )
 
 inherit distutils-r1
```
Comment (mgorny) [inline dev-python/versioneer/versioneer-0.19.ebuild:10]: `${PYTHON}` → `"${EPYTHON}"`

**Review by mgorny — CHANGES_REQUESTED** [review]
Also, please set `DISTUTILS_STRICT_ENTRY_POINTS=1` in yar make.conf and help removing stale `=rdepend`s.

And please make smaller pull requests.

File: dev-python/pgspecial/metadata.xml (hunk 5 above)
Comment (jsmolic) [reply dev-python/pgspecial/metadata.xml:11]: Done, I've adressed the other comments as well

**Comment by mgorny** [comment]
Thanks. I'm going to push this in smaller batches, and get back to you if I hit any issues.

File: dev-python/pgspecial/pgspecial-1.12.1.ebuild [hunk 15]
```diff
@@ -3,7 +3,7 @@
```
Comment (mgorny) [inline dev-python/pgspecial/pgspecial-1.12.1.ebuild:3]: Could you please split pgspecial into a separate PR and attempt to start a server for tests (similarly to how psycopg does it)? Almost all tests are skipped right now.

Comment (jsmolic) [reply dev-python/pgspecial/pgspecial-1.12.1.ebuild:3]: sure
